starfish/core/segmentation_mask.py

```diff
@@ -154,6 +170,43 @@ def from_label_image(
 
         return cls(masks)
 
+    def to_label_image(
+            self,
+            shape: Optional[Tuple[int, ...]] = None,
+            *,
+            ordering: Sequence[Axes] = (Axes.ZPLANE, Axes.Y, Axes.X)
+    ):
+        """Create a label image from the contained masks.
+
+        Parameters
+        ----------
+        shape : Optional[Tuple[int, ...]]
+            Shape of the label image. If ``None``, use maximum index for each axis.
+        ordering : Sequence[Axes]
+            Ordering of the axes. Default is z, y, x.
+        """
+        ordering = [o for o in ordering if self.max_shape[o] > 0]
+
+        max_shape = tuple(self.max_shape[o] for o in ordering)
+
+        if shape is None:
+            shape = max_shape
+        elif np.any(np.less(shape, max_shape)):
+            raise ValueError("shape less than the maximum of the data provided."
+                             "cropping is not supported at this time")
+
+        label_image = np.zeros(shape)
+
+        for i, mask in enumerate(self):
+            s = tuple(slice(mask.coords[o].values[0], mask.coords[o].values[-1] + 1)
+                      for o in ordering if o in mask.coords)
+            # align axes and insert into image
+            label_image[s] = (mask
```

Comment:
here need to just overwrite area where mask is True, not whole bounding box as defined in `s`

Reply:
Would this work

```python
        label_image = np.zeros(shape)

         for i, mask in enumerate(self):
            s = tuple(slice(mask.coords[o].values[0], mask.coords[o].values[-1] + 1)
                      for o in ordering if o in mask.coords)
            # align axes and insert into image
             temp_label_image = np.zeros(shape)

            temp_label_image[s] = (mask.transpose(*[o.value for o in ordering if o in mask.coords]).where(mask == False, i + 1))  # noqa: E712

label_image[temp_label_image > 0] = i + 1
```

Reply:
that would work but it's memory inefficient

Reply:
sure - you can also convert everything to linear indices and then add the appropriate offsets based on the `mask.coords[o].values[0]` values

Reply:
yeah, that's true

Reply:
I've just updated, tell me if that fixes it

Reply:
it's fixed! thanks!